File: src/main/java/com/vaadin/demo/component/switchcomponent/SwitchReadonlyAndDisabled.java
package com.vaadin.demo.component.switchcomponent;

import com.vaadin.flow.component.checkbox.Switch;
import com.vaadin.flow.component.orderedlayout.VerticalLayout;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("switch-readonly-and-disabled")
public class SwitchReadonlyAndDisabled extends VerticalLayout {

    public SwitchReadonlyAndDisabled() {
        setPadding(false); // hidden-source-line

        // tag::snippet[]
        Switch auditLog = new Switch("Audit log retention (90 days)");
        auditLog.setValue(true);
        auditLog.setReadOnly(true);
        auditLog.setHelperText("Included on the Business plan");

        Switch dailyDigest = new Switch("Daily digest");
        dailyDigest.setEnabled(false);

        add(auditLog, dailyDigest);
        // end::snippet[]
    }

    public static class Exporter // hidden-source-line
            extends DemoExporter<SwitchReadonlyAndDisabled> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/switch/switch-readonly-and-disabled.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/switch';
import '@vaadin/vertical-layout';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('switch-readonly-and-disabled')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <vaadin-vertical-layout theme="spacing">
        <!-- tag::snippet[] -->
        <vaadin-switch
          label="Audit log retention (90 days)"
          helper-text="Included on the Business plan"
          readonly
          checked
        ></vaadin-switch>

        <vaadin-switch label="Daily digest" disabled></vaadin-switch>
        <!-- end::snippet[] -->
      </vaadin-vertical-layout>
    `;
  }
}
